public/modules/custom/helfi_rekry_content/helfi_rekry_content.module

```diff
@@ -436,3 +448,95 @@ function helfi_rekry_content_preprocess_block(&$variables) {
     $first_paragraph_gray = 'has-first-gray-bg-block';
   }
 }
+
+/**
+ * Implements hook_page_attachments().
+ */
+function helfi_rekry_content_page_attachments(array &$attachments) : void {
+
+  // Get current entity and entity version.
+  $entity_matcher = \Drupal::service('helfi_platform_config.entity_version_matcher')->getType();
+
+  /** @var \Drupal\Core\Entity\ContentEntityInterface $entity */
+  $entity = $entity_matcher['entity'];
+
+  // No need to continue if current entity doesn't have hero set.
+  if (
+    !$entity instanceof ContentEntityInterface ||
+    $entity->getType() !== 'job_listing'
+  ) {
+    return;
+  }
+
+  /** @var \Drupal\helfi_rekry_content\Entity\JobListing $job_listing */
+  $job_listing = $entity;
+
+  /** @var \Drupal\Core\Datetime\DateFormatterInterface $date_formatter */
+  $date_formatter = Drupal::service('date.formatter');
+
+  $publication_starts_datetime = $job_listing->get('field_publication_starts')->date->getTimestamp();
+  $publication_starts_formatted = $date_formatter->format($publication_starts_datetime, 'html_date');
+
+  /** @var Drupal\Core\Extension\ThemeHandler $theme_handler */
+  $theme_handler = Drupal::service('theme_handler');
+
+  // Get the default share image as structured data logo.
+  if ($theme_handler->themeExists('hdbt')) {
+    $theme = $theme_handler->getTheme('hdbt');
+    $current_language = \Drupal::languageManager()
+      ->getCurrentLanguage(LanguageInterface::TYPE_CONTENT)->getId();
+    $logo_file_name = $current_language === 'sv' ? 'og-global-sv.png' : 'og-global.png';
+
+    /** @var \Drupal\Core\File\FileUrlGeneratorInterface $file_url_generator */
+    $file_url_generator = \Drupal::service('file_url_generator');
+    global $base_url;
+    $logo_image = $file_url_generator->generate("{$theme->getPath()}/src/images/{$logo_file_name}")
```

Comment:
You can generate an absolute URL by replace `->generate()` with `->generateAbsolute()`